substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/SecurityServicesFeature.java

```diff
@@ -309,6 +310,26 @@ public void beforeAnalysis(BeforeAnalysisAccess access) {
             /* Resolve calls to com_sun_security_auth_module_UnixSystem* as builtIn. */
             PlatformNativeLibrarySupport.singleton().addBuiltinPkgNativePrefix("com_sun_security_auth_module_UnixSystem");
         }
+
+        if (isWindows()) {
+            // JNI config required by sunmscapi
+            access.registerReachabilityHandler(SecurityServicesFeature::registerLoadKeysOrCertificateChains,
+                            method(access, "sun.security.mscapi.CKeyStore", "loadKeysOrCertificateChains", String.class));
+            access.registerReachabilityHandler(SecurityServicesFeature::registerGenerateCKeyPair,
+                            method(access, "sun.security.mscapi.CKeyPairGenerator$RSA", "generateCKeyPair", String.class, int.class, String.class));
+            access.registerReachabilityHandler(SecurityServicesFeature::registerCPrivateKeyOf,
+                            method(access, "sun.security.mscapi.CKeyStore", "storePrivateKey", String.class, byte[].class, String.class, int.class),
+                            method(access, "sun.security.mscapi.CSignature", "importECPublicKey", String.class, byte[].class, int.class),
+                            method(access, "sun.security.mscapi.CSignature", "importPublicKey", String.class, byte[].class, int.class));
```

Comment:
Could you factor this out into a helper method, perhaps `registerSunMSCAPIReachabilityHandlers`.